### PR TITLE
add s_gd2

### PR DESCRIPTION
**Pull request by scottgigante**
Checklist

- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [X] Source is from official source
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [X] Build number is 0
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by scottgigante** [comment]
@jxz12 are you happy to be a maintainer of this conda-forge recipe?

**Comment by jxz12** [comment]
> @jxz12 are you happy to be a maintainer of this conda-forge recipe?

Yes I am very happy to do this! I will have a look into the failing checks later this afternoon.

**Comment by scottgigante** [comment]
Great! I'm no expert but it looks like a C++ version issue. Windows 64 build line 2347:

```
 C:\Program Files (x86)\Common Files\Microsoft\Visual C++ for Python\9.0\VC\Include\xlocale(342) : warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc
  ./s_gd2/layout.cpp(22) : error C2039: 'minstd_rand' : is not a member of 'std'
  ./s_gd2/layout.cpp(22) : error C2065: 'minstd_rand' : undeclared identifier
  ./s_gd2/layout.cpp(22) : error C2146: syntax error : missing ';' before identifier 'rng'
  ./s_gd2/layout.cpp(22) : error C3861: 'rng': identifier not found
  ./s_gd2/layout.cpp(24) : error C2143: syntax error : missing ',' before ':'
  ./s_gd2/layout.cpp(25) : error C2143: syntax error : missing ';' before '{'
  ./s_gd2/layout.cpp(27) : error C2039: 'shuffle' : is not a member of 'std'
  ./s_gd2/layout.cpp(27) : error C2065: 'rng' : undeclared identifier
  ./s_gd2/layout.cpp(27) : error C3861: 'shuffle': identifier not found
  ./s_gd2/layout.cpp(30) : error C2143: syntax error : missing ',' before ':'
  ./s_gd2/layout.cpp(30) : error C2530: 't' : references must be initialized
  ./s_gd2/layout.cpp(31) : error C2143: syntax error : missing ';' before '{'
  ./s_gd2/layout.cpp(66) : error C2039: 'unordered_set' : is not a member of 'std'
  ./s_gd2/layout.cpp(66) : error C2065: 'unordered_set' : undeclared identifier
  ./s_gd2/layout.cpp(66) : error C2062: type 'int' unexpected
  ./s_gd2/layout.cpp(70) : error C2143: syntax error : missing ';' before '{'
  ./s_gd2/layout.cpp(71) : error C2065: 'j' : undeclared identifier
  ./s_gd2/layout.cpp(71) : error C2065: 'ij' : undeclared identifier
  ./s_gd2/layout.cpp(71) : error C2143: syntax error : missing ',' before ';'
  ./s_gd2/layout.cpp(76) : error C2143: syntax error : missing ';' before '{'
  ./s_gd2/layout.cpp(81) : error C2143: syntax error : missing ';' before '}'
  ./s_gd2/layout.cpp(82) : error C2143: syntax error : missing ';' before '}'
  ./s_gd2/layout.cpp(84) : error C2143: syntax error : missing ';' before '}'
  ./s_gd2/layout.cpp(88) : error C2146: syntax error : missing ',' before identifier 'I'
  ./s_gd2/layout.cpp(88) : error C2146: syntax error : missing ',' before identifier 'J'
  ./s_gd2/layout.cpp(88) : error C2143: syntax error : missing ',' before ')'
  ./s_gd2/layout.cpp(89) : error C2143: syntax error : missing ';' before '{'
  ./s_gd2/layout.cpp(90) : error C2143: syntax error : missing ',' before ')'
  ./s_gd2/layout.cpp(99) : error C2143: syntax error : missing ';' before '{'
  ./s_gd2/layout.cpp(100) : error C2143: syntax error : missing ',' before ')'
  ...
```

**Comment by scottgigante** [comment]
This may be relevant: https://conda-forge.readthedocs.io/en/latest/buildwin.html

**Comment by jxz12** [comment]
I think you are correct that the c++ version is the issue, and I'm pretty sure the issue is that it's ignoring the -std=c++11 flag:
```
  C:\Program Files (x86)\Common Files\Microsoft\Visual C++ for Python\9.0\VC\Bin\amd64\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -IC:\Miniconda\conda-bld\s_gd2_1575502074245\_h_env\lib\site-packages\numpy\core\include -IC:\Miniconda\conda-bld\s_gd2_1575502074245\_h_env\include -IC:\Miniconda\conda-bld\s_gd2_1575502074245\_h_env\PC /Tp./s_gd2/layout.cpp /Fobuild\temp.win-amd64-2.7\Release\./s_gd2/layout.obj -std=c++11
  cl : Command line warning D9002 : ignoring unknown option '-std=c++11'
```
This could be fixed by forcing it to use the gcc compiler (is that possible on Windows? maybe through mingw?). A better solution would be I think to dynamically change the setuptools `extra_compile_args` depending on the compiler, to flag to `/std:c++11` when the cl.exe compiler is used on windows. But I'm not sure if that's possible either?

**Comment by scottgigante** [comment]
@conda-forge/help-python-c does anyone know how to specify `-std='c++11'` when `g++` is used and `/std:c++11` when `cl.exe` is used to compile extensions with `setuptools`?

**Comment by jxz12** [comment]
I have created a new release on the main s_gd2 repo with all the c++11 features removed. After another look I think this might be the only way to pass the windows build because it uses the visual studio 2008 compiler for python 2.7:
```
Computed that there are 3 distributions to build from 1 recipes
Resolved dependencies, will be built in the following order:
    s_gd2-1.5-python_3.6-cxx_compiler_vs2015-on-win-64
    s_gd2-1.5-python_3.7-cxx_compiler_vs2015-on-win-64
    s_gd2-1.5-python_2.7-cxx_compiler_vs2008-on-win-64
```
and c++11 would not be available on there without a time machine! The new release should fix this nicely, but I wonder if there's an option for skipping python 2.7 on windows? I can't imagine that every package on conda is restricted to archaic pre-11 c++, and it would be useful to know how to get around that in the future.

**Comment by scottgigante** [comment]
There are a few ways to do that, yes.

1. I think you can choose requirements by platform, e.g.
```
host:
  - python # [unix]
  - python >=3.5 # [win]
```
or alternatively just drop python 2 support, at least in conda, since Python 2 will hit official end of life in three weeks.

**Comment by scottgigante** [comment]
@jxz12 looks like we've still got some C++ errors: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=101598

**Comment by jxz12** [comment]
More c++11 errors! I forgot to test compilation of the sparse.cpp things under c++98, and one small function has caught me. Will fix tomorrow.

**Comment by jxz12** [comment]
So I double checked my tests and it turns out that I did run the right tests without c++11, but my compiler didn't downgrade the ISO standard correctly... 

I need to integrate a new random number generator anyway, so I will make package this c++ fix along with that.

File: recipes/example/meta.yaml [hunk 1]
```diff
@@ -1,83 +0,0 @@
-# Note: there are many handy hints in comments in this example -- remove them when you've finalized your recipe
```
Comment (scopatz) [inline recipes/example/meta.yaml]: Please restore the example

File: recipes/s_gd2/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+  host:
+    - python
+    - pip
+    - numpy >=1.16.0
```
Comment (scopatz) [inline recipes/s_gd2/meta.yaml]: Please see https://conda-forge.org/docs/maintainer/knowledge_base.html?#building-against-numpy for linking against numpy

Comment (scottgigante) [reply recipes/s_gd2/meta.yaml]: Is this right?

Comment (scopatz) [reply recipes/s_gd2/meta.yaml]: Yep!

**Comment by scopatz** [comment]
Is this ready to go in?

**Comment by jxz12** [comment]
Yes please!

**Comment by scottgigante** [comment]
Thanks!

**Pull request merged**